HIP/hip-583.md

```diff
@@ -10,220 +10,518 @@ status: Accepted
+It **is** possible to specify an `account-num alias` as the _target_ of a `CryptoTransfer` transaction. Indeed, for
+Ethereum-native wallets, this is the only way to transfer value to Hedera accounts that don't have an
+`evm address alias`. However, it is **impossible** to auto-create an account using the `account-num alias`. If the
+account does not exist, the transfer will fail.
 
-### Add  setAlias to `CryptoCreate`
+##### `key alias`
 
-![auto-create-with-key.png](../assets/hip-583/auto-create-with-key.png)
+The `key alias` is a protobuf-encoded `Key` object. We only support `ED25519` and `ECDSA_secp256k1` keys at this time.
+When an attempt is made to transfer value into an account using `CryptoTransfer` with an alias, we check whether it is
+a `key alias` by checking if there are more than 20 bytes. We then:
 
-Add alias bytes property to the `CryptoCreateTransactionBody` [proto file](https://github.com/hashgraph/hedera-protobufs/blob/main/services/crypto_create.proto)
+1. Deserialize the protobuf object. If the object is not a protobuf object, we reject the transaction.
+2. If the protobuf object is not an `ED25519` or `ECDSA_secp256k1` `Key`, we reject the transaction.
+3. If the `Key` is an `ED25519` key:
+    1. If the alias is already in use, we use that account and do not create a new one
+    2. If the alias is not in use, we create a new account and use the given key as the key on the account
+4. If the `Key` is an `ECDSA_secp256k1` key:
+    1. If the alias is already in use, we use that account and do not create a new one
+    2. If the derived `evm address` from the key is already in use, we use that account and do not create a new one
+    3. If the alias is not in use, we create a new account and use the given key as the key on the account.
 
-```proto
-message CryptoCreateTransactionBody {
-	...
-	
-	/**
-   * The bytes to be used as the account's alias. The bytes will be 1 of 2 options. It will be the serialization
-   * of a protobuf Key message for an ED25519/ECDSA primitive key type. If the account is ECDSA based it may also be the public-address, calcluated as the last 20 bytes of the keccak-256 of the ECDSA primitive key.
-   * Currently only primitive key bytes are supported as the key for an account with an alias. ThresholdKey, KeyList, ContractID, and delegatable_contract_id are not supported.
-   *
-   * At most only one account can ever have a given alias on the network.
-   *
-   * If a transaction creates an account using an alias, any further crypto transfers to that alias will simply be deposited
-   * in that account, without creating anything, and with no creation fee being charged.
-	 * If an account with an alias is deleted the alias will be freed up for use.
-   */
-	bytes alias = 18;
-}
-```
+If we do create a new account from either `CryptoCreate` or `CryptoTransfer`, and the `alias` is an `ECDSA_secp256k1`
```

Comment:
The mention of `CryptoCreate` when it comes to `ECDSA_secp256k1` should be removed.